jaeger-micrometer/src/main/java/com/uber/jaeger/micrometer/MicrometerStatsFactory.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2017, The Jaeger Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.uber.jaeger.micrometer;
+
+import static io.micrometer.core.instrument.Metrics.counter;
+import static io.micrometer.core.instrument.Metrics.gauge;
+import static io.micrometer.core.instrument.Metrics.timer;
+
+import com.uber.jaeger.metrics.Counter;
+import com.uber.jaeger.metrics.Gauge;
+import com.uber.jaeger.metrics.StatsFactory;
+import com.uber.jaeger.metrics.Timer;
+import io.micrometer.core.instrument.ImmutableTag;
+import io.micrometer.core.instrument.Tag;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class MicrometerStatsFactory implements StatsFactory {
+  @Override
+  public Counter createCounter(final String name, final Map<String, String> tags) {
+    log.trace("Creating counter {}", name);
+    return new Counter() {
+      private final io.micrometer.core.instrument.Counter counter = counter(name, translateTags(tags));
+
+      @Override
+      public void inc(long delta) {
+        log.trace("Incrementing counter {} by {}", name, delta);
```

Comment:
please remove logging on metrics

Reply:
I can remove it, but this call is a noop after a few cycles, as hotspot-based JVMs (and probably others) will just use branch prediction to skip it once it sees that it's never executed.

Or are you concerned about something else?